hydra-cluster/src/CardanoNode.hs

```diff
@@ -125,6 +125,28 @@ getCardanoNodeVersion :: IO String
 getCardanoNodeVersion =
   readProcess "cardano-node" ["--version"] ""
 
+-- | Tries to reach for an existing cardano-node running on default state directory.
+findRunningCardanoNode :: FilePath -> KnownNetwork -> IO (Maybe RunningNode)
+findRunningCardanoNode stateDirectory knownNetwork = do
+  let args'@CardanoNodeArgs{nodeSocket} = defaultCardanoNodeArgs
+  let nodeSocketPath = File nodeSocket
+  maybeSocketPath <- doesFileExist $ unFile nodeSocketPath
+  if maybeSocketPath
+    then do
+      let
+        knownNetworkId = case knownNetwork of
+          Mainnet -> Api.Mainnet
+          Preproduction -> Api.Testnet (Api.NetworkMagic 1)
+          Preview -> Api.Testnet (Api.NetworkMagic 2)
+          Sanchonet -> Api.Testnet (Api.NetworkMagic 4)
+      try (queryTip knownNetworkId nodeSocketPath)
+        >>= \case
+          Right _ -> pure Nothing
+          Left (_ :: SomeException) -> do
+            n@RunningNode{networkId} <- mkRunningNode stateDirectory nodeSocketPath args'
```

Comment:
Should: not need to path both `nodeSocketPath` and `args'` to `mkRunningNode`

The arguments do include the node socket file name, so this is redundant.

You could call `RunningNode{networkId, nodeSocket} <- mkRunningNode ...` in the very beginning and using that `try (queryTip networkId nodeSocket) >>= \case` here. This will simplify the overall call sequence.